src/routes/Bridge.js

```diff
@@ -0,0 +1,676 @@
+                        console.log('Estimation ', result.estimation)
+                        const hash = await this.sdk.submitGatewayBatch()
+                        console.log('Transaction submitted ', hash)
+                        await this.setState({
+                          loading_msg: 'Transaction Submitted'
+                        })
+                        let flag = 1
+                        while (flag) {
+                          let submit_gateway_batch = await this.sdk.getGatewaySubmittedBatch(
+                            {
+                              hash: hash.hash
+                            }
+                          )
+                          if (
+                            submit_gateway_batch.state === 'Sending' &&
+                            flag === 1
+                          ) {
+                            flag = 0
+                            console.log(submit_gateway_batch.transaction)
+                            await this.getBalance()
+                            await this.setState({
+                              loading: false,
+                              loading_msg: ''
+                            })
+                            alert(
+                              'Transferred Successfully ' +
+                                submit_gateway_batch.transaction.hash
+                            )
+                            return
+                          } else {
+                            await Sdk.sleep(5)
+                          }
+                        }
+                      })
+                      .catch(error => {
+                        this.setState({
+                          loading: false,
+                          loading_msg: ''
+                        })
+                        console.log(
+                          'Transaction estimation failed with error ',
+                          error
+                        )
+                      })
+                  } else {
+                    alert('Invalid entry on withdrawal')
+                    console.log('invalid entry')
+                    await this.setState({
+                      loading: false,
+                      loading_msg: ''
+                    })
+                  }
+                }
+              })
+            } else {
+              await Sdk.sleep(5)
+              // console.log("i", i++)
+            }
+          }
+        } else {
+          alert('Not Enough Balance')
+          this.setState({
+            loading: false,
+            loading_msg: ''
+          })
+        }
+      } else {
+        this.setState({
+          loading: false,
+          loading_msg: ''
+        })
+        console.log(this.state.errors)
+        alert('Minimum amount accepted by Token Bridge is 10')
+      }
+    } catch (err) {
+      this.setState({
+        loading: false,
+        loading_msg: ''
+      })
+      console.log(err)
+    }
+  }
+
+  Dai_to_xDai = async () => {
```

Comment:
seems some overwrapping logic with xDAI_to_DAI. Can you generalise so it makes easier to do $MATIC_l1 -> $Matic_l2 . We could also integrate with something like https://li.finance to do cross chain swap so users can deposit with any token they have on their wallet